### DISCUSSION TITLE
Adding new land coverage constraints based on "terrain profile" through Atlite

### DISCUSSION THREAD

**Discussion by pz-max** [pypsa-meets-earth/pypsa-earth#26] [Ideas]
Context:
Atlite can exclude i.e. offshore wind potentials through looking at bathymetry data (deepness of the sea). Similar excluded are also areas that are currently under protection. 

Idea:
What about excluding also areas for certain technologies based on the slope/ or terrain profile on land? This could for instance lead to exclusion of wind potentials around mountains/steep hills or the exclusion of solar potentials on the north/south slope of mountains/hills (depending on the location -> south or north of the equator)

Tool:
GemGIS seems to be a useful Python-based open-source tool for that purpose. It is described as spatial data processing for geological modelling. Check out [GemGIS](https://github.com/cgre-aachen/gemgis).

**Comment by pz-max** [comment]
@euronion not for our prototype but maybe smt for later?

**Reply by euronion** [reply]
Yes, I think that's a good idea!
Maybe we don't even need the additional software dependency.

**Comment by pz-max** [comment]
- Slope can be generated on a global scale. E.g. use Shuttle Radar Topography Mission (STRM) at 30x30 m2 resolution and calculate the slope:
- `Slope angle = np.arctan(difference in elevation between two neighbouring rasters / "Bird view" distance between two points`). FYI, the slope angle for one rectangular raster cell can generate slope angles in all directions (north, south, etc. which can be useful/ or over-engineered)
- Exclusion condition per technology needs some thoughts

**Comment by euronion** [comment]
If someone is interested in tackling this:

Slope can also be generated from GEBCO data using `gdal`. An implementation similiar on how we could derive slope from the GEBCO data and include into the workflow can be found here

https://github.com/euronion/trace/blob/replace-gegis/rules/renewables.smk

and here

https://github.com/euronion/trace/blob/replace-gegis/actions/build_potentials_and_profiles.py.ipynb

Slope values to restrict RES expansion to can be found e.g. in this IRENA report:

https://www.irena.org/-/media/Files/IRENA/Agency/Publication/2015/IRENA-LBNL_Africa-RE-_CEC_2015.pdf

or in the IRENA MSR repository directly:

https://github.com/bhussain89/Model-Supply-Regions-MSR-Toolset

**Reply by davide-f** [reply]
Thanks @euronion for the suggestion! since you highlighted, would you like to create the issue?

**Reply by euronion** [reply]
Sure! #377
